bw2data/project.py

```diff
@@ -571,23 +573,62 @@ def signal_dispatcher(
     """Not sure why this is necessary, but fails silently if call `add_revision` directly"""
     from bw2data import revisions
 
-    delta = revisions.generate_delta(old, new, operation)
+    try:
+        delta = revisions.generate_delta(old, new, operation)
+    except NoRevisionNeeded:
+        return
```

Comment:
Why do we raise if we then just ignore it? Would it be better for the execution flow to just `return None` and act here accordingly?